src/gui/progressbardelegate.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2020 Prince Gupta <jagannatharjun11@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+ #pragma once
+
+ #include <QStyledItemDelegate>
+ #include <QProgressBar>
+
+class QStyleOptionProgressBar;
+
+class ProgressBarDelegate : public QStyledItemDelegate
+{
+public:
+    ProgressBarDelegate(int progressColumn, int dataRole, QObject *parent = nullptr);
+
+protected:
+    virtual void initProgressStyleOption(QStyleOptionProgressBar &option, const QModelIndex &index) const;
+
+private:
+    void paint(QPainter *painter, const QStyleOptionViewItem &option, const QModelIndex &index) const;
+
+    const int m_progressColumn;
+    const int m_dataRole;
+    QProgressBar m_dummyProgressBar; // for painting progressbar with stylesheet option, a dummy progress bar is required
```

Comment:
Please move comment into separate line before member declaration. Otherwise it makes line too long.